clients/src/main/java/org/apache/kafka/clients/admin/internals/ListOffsetsHandler.java

```diff
@@ -93,13 +94,38 @@ ListOffsetsRequest.Builder buildBatchedRequest(int brokerId, Set<TopicPartition>
             .stream()
             .anyMatch(key -> offsetTimestampsByPartition.get(key) == ListOffsetsRequest.MAX_TIMESTAMP);
 
+        boolean requireEarliestLocalTimestamp = keys
+                .stream()
+                .anyMatch(key -> offsetTimestampsByPartition.get(key) == ListOffsetsRequest.EARLIEST_LOCAL_TIMESTAMP);
+
         boolean requireTieredStorageTimestamp = keys
```

Comment:
Could you please move this "helper" to `ListOffset.Builder`? normally, we should not use this helper as it requires to "loop" data. However, if callers have no idea about the "timestamp" type, this helper can find out that for them.

Reply:
Sounds good 😃